chore: adding veneer implementation for DataClientWrapper

rahulKQL opened the pull request:
This PR adds veneer specific implementation for `DataClientWrapper`. Also, adding ResultRowAdapter which helps in converting `ReadRowsResponse` to HBase's `Result`.

rahulKQL committed:
chore: adding veneer implementation for DataClientWrapper

This PR adds veneer specific implementation for `DataClientWrapper`. Also, adding ResultRowAdapter which helps in converting `ReadRowsResponse` to HBase's `Result`.

rahulKQL committed:
Marked DataClientVeneerApi as internal only

dmitry-fa commented on bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/read/ResultRowAdapter.java:
```diff
@@ -0,0 +1,181 @@
+  }
+
+  public static class ResultRowBuilder implements RowBuilder<Result> {
+    private byte[] currentKey;
```
should this class be public? If yes, you need to provide tests for it and class description doc.
But I guess you can make it private (I can be wrong)

dmitry-fa commented on bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/read/ResultRowAdapter.java:
```diff
@@ -0,0 +1,181 @@
+     *   <li>qualifiers in each family cluster is sorted lexicographically
+     *   <li>then descending by timestamp
+     * </ul>
+     *
```
you need to rephrase the order, it's not clear at all.
Probably you meant: Cells with the the same family are sorted according to:
- by clustered/not clustered
- by qualifiers (lexicographically)
- by timestamp (descending)

perhaps you can skip this paragraph, you have the explanation further

dmitry-fa commented on bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/read/ResultRowAdapter.java:
```diff
@@ -0,0 +1,181 @@
+     */
+    @Override
+    public void finishCell() {
+      if (!Objects.equals(this.family, this.previousFamily)) {
```
Did you check that cell has been started?

dmitry-fa commented on bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/read/ResultRowAdapter.java:
```diff
@@ -0,0 +1,181 @@
+     * </ul>
+     *
+     * The end result will be that {@link RowCell} are ordered as:
+     *
```
The end result => The final result

dmitry-fa commented on bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/read/ResultRowAdapter.java:
```diff
@@ -0,0 +1,181 @@
+     */
+    @Override
+    public Result finishRow() {
+      if (totalCellCount == 0) {
```
Did you check that Row has been started before?
Did you check that there is no 'started' cells?

dmitry-fa commented on bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/read/ResultRowAdapter.java:
```diff
@@ -0,0 +1,181 @@
+
+    @Override
+    public void startCell(
+        String family, ByteString qualifier, long timestamp, List<String> labels, long size) {
```
What is another cell has been already started but not finished yet?

rahulKQL committed:
chore: address feedback comments

 - Renamed and Moved RowResultAdapter within veneer package
 - created custom class to serve getKey() and isScanMarkerRow()
 - optimized cellValue assignment
 - removed unnecessary label check and deduping

rahulKQL committed:
chore: address feedback comments

 - updated & optimized RowResultAdapter according to suggested changes.
 - updated existing and added new test cases

rahulKQL committed:
chore: addressed feedback comment for RowResult factory methods

rahulKQL committed:
chore: to trigger CI job
